lib/client.js

```diff
@@ -302,6 +302,64 @@ class Client {
     return Promise.resolve(this.__dataApiClient)
   }
 
+  /**
+   * Returns a decoded Avro Promise given the proper schema name and data
+   * @param {String} schemaName
+   * @param {Object|String} data
+   * @returns {Promise}
+   */
+  decodeData (schemaName, data) {
+    if (!schemaName || typeof schemaName !== 'string' || schemaName.trim() === '') {
+      log.debug('decodeData() schemaName parameter not defined');
+      return null;
+    }
+
+    if (!data) {
+      log.debug('decodeData() data parameter not defined or empty');
+      return null;
+    }
+
+    return this.dataApiClient().then((api) => {
+      return api.get(`current-schemas/${schemaName}`, { authenticate: false }).then((schema) => {
+        log.debug('Got schema: ' + `current-schemas/${schemaName}`, schema);
+        const avroType = avsc.parse(JSON.parse(schema.schema));
+
+        try {
+          const decode = (data) => this.decodeAvroBufferString(data, avroType);
+          return Array.isArray(data) ? data.map(decode) : decode(data);
+        } catch (e) {
+          throw new AvroValidationError(avroType, data)
+        }
+      });
+    });
+  }
+
+  /**
+   * Returns a decoded Avro Object from a given encoded Buffer
+   * @param {String} bufferString
+   * @param {Object} avroObject
+   * @param {String} encodeType - default encode type base64.
+   * @returns Returns a deserialized buffer
+   */
+  decodeAvroBufferString (bufferString, avroObject, encodeType = 'base64') {
+    if (!bufferString) {
+      log.debug('Error: decodeAvroBufferString() bufferString parameter not defined or empty');
+      return;
+    }
+
+    if (typeof avroObject !== 'object') {
```

Comment:
Will avsc.parse return anything other than an object? Wondering if this check is really necessary?